src/nupic/algorithms/sdr_classifier.py

```diff
@@ -0,0 +1,393 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+Implementation of a SDR classifier.
+
+The SDR classifier takes the form of a single layer classification network
+that takes SDRs as input and outputs a predicted distribution of classes.
+"""
+
+from collections import deque
+
+import numpy
+
+
+
+class SDRClassifier(object):
+  """
+  The SDR Classifier accepts a binary input pattern from the
+  level below (the "activationPattern") and information from the sensor and
+  encoders (the "classification") describing the true (target) input.
+
+  The SDR classifier maps input patterns to class labels. There are as many
+  output units as the number of class labels or buckets (in the case of scalar
+  encoders). The output is a probabilistic distribution over all class labels.
+
+  During inference, the output is calculated by first doing a weighted summation
+  of all the inputs, and then perform a softmax nonlinear function to get
+  the predicted distribution of class labels
+
+  During learning, the connection weights between input units and output units
+  are adjusted to maximize the likelihood of the model
+
+  The SDR Classifier is a variation of the previous CLAClassifier which was
+  not based on the references below.
+
+  Example Usage:
+
+  c = SDRClassifier(steps=[0], alpha=0.1, actValueAlpha=0.1, verbosity=0)
```

Comment:
I think it would be better to use steps=[1] in this example - that's the most common case.